### PR TITLE
Add Qos test for bridge type interface

### PR DESCRIPTION
**Pull request by yalzhang**
Test Qos for bridge type interface and networks. Move the tc rule
check function from the iface_network.py into avocado-vt for sharing.
Depends on https://github.com/avocado-framework/avocado-vt/pull/3124

Signed-off-by: Yalan <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
new add 3 cases:
VT type_specific.io-github-autotest-libvirt.virtual_network.iface_bridge.test_qos.start.shared_physical_network
VT type_specific.io-github-autotest-libvirt.virtual_network.iface_bridge.test_qos.hotplug_iface.shared_physical_network
VT type_specific.io-github-autotest-libvirt.virtual_network.iface_bridge.test_qos.network.shared_physical_network

all cases related with the change passed.
$avocado run --vt-type libvirt virtual_network.iface_network  virtual_network.iface_bridge
JOB ID     : 722fdafbe9c0611bde31d2b3e18e8d2e8be4e64e
JOB LOG    : /root/avocado/job-results/job-2021-07-06T10.30-722fdaf/job.log
 (01/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.rom_test.negative.enable_no_bar_off: PASS (16.57 s)
...
(06/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.dnsmasq_test.net_host_ip: ERROR: Failed to install bind-utils on guest (409.88 s)
...
 (29/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_pxe.default: CANCEL: Please provide initrd/vmlinuz URL (59.02 s)
 (30/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_pxe.boot_failure.virtio: CANCEL: Please provide initrd/vmlinuz URL (20.66 s)
 (31/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_pxe.boot_failure.e1000: CANCEL: Please provide initrd/vmlinuz URL (15.25 s)
 (32/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_pxe.boot_failure.rtl8139: CANCEL: Please provide initrd/vmlinuz URL (14.70 s)
 (33/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_pxe.boot_failure.e1000e: CANCEL: Please provide initrd/vmlinuz URL (12.64 s)
 (34/55) type_specific.io-github-autotest-libvirt.virtual_network.iface_network.net_guest: ERROR: Failed to install libvirt package on guest (396.54 s)
....
RESULTS    : PASS 48 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 5
JOB TIME   : 5030.53 s

**Comment by yalzhang** [comment]
virtual_network.iface_bridge.test_qos.network.shared_physical_network  
to automate RHEL-187270

virtual_network.iface_bridge.test_qos.start.shared_physical_network
virtual_network.iface_bridge.test_qos.hotplug_iface.shared_physical_network
to automate RHEL-254166

**Comment by yalzhang** [comment]
@chloerh @Yingshun Please help to review, Thank you!

**Comment by Yingshun** [comment]
@yalzhang rebase,please!

**Comment by chloerh** [comment]
Conflicting

**Comment by yalzhang** [comment]
> Conflicting

updated.

**Comment by yalzhang** [comment]
> @yalzhang rebase,please!

Updated, Thank you!

**Review by kamilvarga — APPROVED** [review]
LGTM.

File: libvirt/tests/src/virtual_network/iface_bridge.py [hunk 1]
```diff
@@ -311,11 +330,31 @@ def check_net_functions(guest_ip, ping_count, ping_timeout, guest_session, host_
...
+                else:
+                    iface_mac = vm_xml.VMXML.get_first_mac_by_name(vm1_name)
+                    iface_name = libvirt.get_ifname_host(vm1_name, iface_mac)
+                    logging.debug("Test inbound:")
+                    res1 = utils_net.check_class_rules(iface_name, "1:1", ast.literal_eval(iface_inbound))
+                    logging.debug("Test outbound:")
+                    res2 = utils_net.check_filter_rules(iface_name, ast.literal_eval(iface_outbound))
+                if not res1 or not res2:
+                    test.fail("Qos test fail!")
+            except AssertionError:
```
Comment (Yingshun) [inline libvirt/tests/src/virtual_network/iface_bridge.py:349]: AssertionError exception will be catch up in utils_net.check_filter_rules(), so do you still need this line?

Comment (yalzhang) [reply libvirt/tests/src/virtual_network/iface_bridge.py]: updated and delete the try... block. I'm not sure is that ok?

**Review by Yingshun — APPROVED** [review]

**Review by chloerh — APPROVED** [review]

**Comment by chloerh** [comment]
@Yingshun @kamilvarga Thank you for helping review. Let's merge it.
Dependency has been merged.

**Pull request merged**